modules/delegates.js

```diff
@@ -687,17 +652,19 @@ Delegates.prototype.updateForgingStatus = function(
  * Gets delegate list based on input function by vote and changes order.
  *
  * @param {number} round
- * @param {function} source - Source function for get delegates
  * @param {function} cb - Callback function
  * @param {Object} tx - Database transaction/task object
  * @returns {setImmediateCallback} cb, err, truncated delegate list
  * @todo Add description for the params
  */
-Delegates.prototype.generateDelegateList = function(round, source, cb, tx) {
+Delegates.prototype.getForgersList = function(round, cb, tx) {
 	// Set default function for getting delegates
-	source = source || __private.getKeysSortByVote;
-
-	source((err, truncDelegateList) => {
+	const lastBlockRound = slots.calcRound(modules.blocks.lastBlock.get().height);
+	const source =
+		lastBlockRound > round
```

Comment:
As I described in previous comments this condition `lastBlockRound > round` will never be `true` because it's always `lastBlockRound <= round`.